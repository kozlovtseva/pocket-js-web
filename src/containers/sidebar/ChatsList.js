import React from 'react'
import { Link } from 'react-router-dom'


const ChatsList = () => (
    <div>
      <Link to='/chats'>Чаты</Link>
    </div>
)

export default ChatsList


