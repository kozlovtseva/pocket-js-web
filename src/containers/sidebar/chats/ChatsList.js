import React from 'react';

import {setActiveChat, getGroupProfile} from '../../../store/actions/index';
import {connect} from 'react-redux';

import ChatsItem from './ChatsItem';

import styles from './ChatsList.module.css';

export class ChatsList extends React.Component {

    clickTimeout = null;
    click_count = 1;

    handleClicks = (id) => {
        if (this.clickTimeout !== null) {
            this.click_count++;
        } else {
            this.clickTimeout = setTimeout(() => {
                //один клик
                if(this.click_count === 1){
                    this.props.dispatch(setActiveChat(id, 1));
                //два клика
                } else {
                    this.props.dispatch(getGroupProfile(id)).then(() => {
                        this.props.openProfile();
                    })
                }
                this.click_count = 1;
                clearTimeout(this.clickTimeout);
                this.clickTimeout = null
            }, 300)
        }
    }

    render(){
        let chats = this.props.chats.map((chat, index) => {
            return <ChatsItem key = {index}
                              handleClicks = {this.handleClicks}
                            {...chat}/>
        }) : [];
        return (
            <>
                <div className = {styles.List}>
                    { chats }
                </div>
                <div className = {styles.ButtonsBlock}>
                    <button className = {styles.Button} onClick = {this.props.createGroup} >
                        <div>
                            <i className = {styles.ButtonIcon + ' fas fa-check'}/>
                        </div>
                        <span className = {styles.ButtonText}>Создать группу</span>
                    </button>
                    <button className = {styles.Button} onClick = {this.props.searchGroup} >
                        <div className = {styles.ButtonIcon}>+</div>
                        <span className = {styles.Text}> Добавить группу</span>
                    </button>
                </div>
            </>
        );
    }
}

export default connect()(ChatsList)