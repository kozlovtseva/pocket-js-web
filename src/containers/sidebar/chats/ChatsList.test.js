import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { ChatsList } from './ChatsList';
// import undecorated component -
// Note the curly braces: grab the named export instead of default export

const props = [
  {
    id: '5ca0c0eb9839f35abe9d4836',
    preview: 'some text',
  },
  {
    id: '5ca0c0eb9839f35abe9d4837',
    preview: 'another text',
  },
];
describe('render ChatsList', () => {
  it('render correctly ChatsList component', () => {
    const ChatsListComponent = renderer.create(<ChatsList />).toJSON();
    expect(ChatsListComponent).toMatchSnapshot();
  });
  it('render correctly Chats items', () => {
    const wrapper = shallow(<ChatsList chats={props} />);
    expect(wrapper.at(0).find('ChatsItem')).toHaveLength(2);
  });
});

describe('invoking the "handleClicks" method from component instance', () => {
  it('click_count should be 1 by default and updated after the second invoking', () => {
    const wrapper = shallow(<ChatsList chats={props} />);
    const instance = wrapper.instance();
    expect(instance.click_count).toBe(1);
    instance.clickTimeout = 200;
    instance.handleClicks();
    expect(instance.click_count).toBe(2);
  });
});
