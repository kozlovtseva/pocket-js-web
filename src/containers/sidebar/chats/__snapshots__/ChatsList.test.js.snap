// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ChatsList render correctly ChatsList component 1`] = `
Array [
  <div
    className="List"
  />,
  <div
    className="ButtonsBlock"
  >
    <button
      className="Button"
    >
      <div>
        <i
          className="ButtonIcon fas fa-check"
        />
      </div>
      <span
        className="ButtonText"
      >
        Создать группу
      </span>
    </button>
    <button
      className="Button"
    >
      <div
        className="ButtonIcon"
      >
        +
      </div>
      <span
        className="Text"
      >
         Добавить группу
      </span>
    </button>
  </div>,
]
`;
